staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -452,15 +569,23 @@ func (r *crdHandler) createCustomResourceDefinition(obj interface{}) {
 	// this could happen if the create event is merged from create-update events
 	storageMap := r.customStorage.Load().(crdStorageMap)
 	oldInfo, found := storageMap[crd.UID]
-	if !found {
-		return
+
+	var tearDownFinishedCh <-chan struct{}
+	if found {
+		if apiequality.Semantic.DeepEqual(&crd.Spec, oldInfo.spec) && apiequality.Semantic.DeepEqual(&crd.Status.AcceptedNames, oldInfo.acceptedNames) {
+			klog.V(6).Infof("Ignoring customresourcedefinition %s create event because a storage with the same spec and accepted names exists",
+				crd.Name)
+			return
+		}
+		// Update storage version with the latest info in the watch event
+		tearDownFinishedCh = r.removeStorageLocked(crd.UID)
 	}
-	if apiequality.Semantic.DeepEqual(&crd.Spec, oldInfo.spec) && apiequality.Semantic.DeepEqual(&crd.Status.AcceptedNames, oldInfo.acceptedNames) {
-		klog.V(6).Infof("Ignoring customresourcedefinition %s create event because a storage with the same spec and accepted names exists",
-			crd.Name)
-		return
+	if utilfeature.DefaultFeatureGate.Enabled(features.StorageVersionAPI) &&
+		utilfeature.DefaultFeatureGate.Enabled(features.APIServerIdentity) {
+		processedCh := make(chan struct{})
+		ctx := apirequest.NewContext()
+		r.storageVersionManager.UpdateStorageVersion(ctx, crd, tearDownFinishedCh, processedCh)
```

Comment:
if the call to `updateCRDStorageVersion` fails, why should we proceed and release the lock without an indication?

I think this is safe because we're removing the storage being used (not actually adding a new one), thus its the next create that mattters.  But if that's the case, a comment is required.

Reply:
Now that UpdateStorageVersion() returns an error, added a logline for the failure case.

Also addded a comment as per suggestion, but unsure if that's what you meant.